### ISSUE TITLE
Is there any way that the guest OS can communicate with each other?

### ISSUE THREAD

**Issue by wjhtinger** [bao-project/bao-hypervisor#13]
Just like some IPC that guest OS can communicate

**Comment by josecm** [comment]
Yes there is. You can create shared memory region and link two (or more) guests via that same region. Guests connected to a shared memory region might have a number of associated doorbell virtual interrupts. A guest can ask the hypervisor (via hypercall) to broadcast an event for a given shared memory region which will result in the other guest receiving the interrupt.

The [example configuration](https://github.com/bao-project/bao-hypervisor/blob/61832198ccb602da37b48a7184a7720e5dace765/configs/example/config.c) shows the basic idea how this is setup. The [linux-freertos demo](https://github.com/bao-project/bao-demos/tree/master/demos/linux%2Bfreertos) showcases this functionality by having a toy example of a Linux and a FreeRTOS guest communicating through these mechanism. If you look both through the demo configuration and guest code, you should be able to understand how to use it. But if you need any more details please feel free to ask.

**Comment by wjhtinger** [comment]
> @wjhtinger were you able to use the inter-vm communciation facilities?

Yes, I have looked the  inter-vm demo in freertos&Linux, It is OK. I will try it, thank you!
